datafusion/functions-aggregate/src/approx_percentile_cont.rs

```diff
@@ -133,18 +133,30 @@ impl Default for ApproxPercentileCont {
 impl ApproxPercentileCont {
     /// Create a new [`ApproxPercentileCont`] aggregate function.
     pub fn new() -> Self {
-        let mut variants = Vec::with_capacity(NUMERICS.len() * (INTEGERS.len() + 1));
-        // Accept any numeric value paired with a float64 percentile
-        for num in NUMERICS {
-            variants.push(TypeSignature::Exact(vec![num.clone(), DataType::Float64]));
-            // Additionally accept an integer number of centroids for T-Digest
-            for int in INTEGERS {
-                variants.push(TypeSignature::Exact(vec![
-                    num.clone(),
-                    DataType::Float64,
-                    int.clone(),
-                ]))
-            }
+        let percentile_coercion = Coercion::new_implicit(
+            TypeSignatureClass::Native(logical_float64()),
+            vec![
+                TypeSignatureClass::Integer,
+                TypeSignatureClass::Float,
+                TypeSignatureClass::Decimal,
+            ],
+            NativeType::Float64,
+        );
+
+        // Value must be numeric (excluding Decimal) as the accumulator currently only
+        // supports integers and floats. Percentile may be any numeric literal that can
+        // be coerced to Float64 (to support parse_float_as_decimal).
+        let mut variants = Vec::with_capacity(4);
+        for value_class in [TypeSignatureClass::Integer, TypeSignatureClass::Float] {
+            variants.push(TypeSignature::Coercible(vec![
+                Coercion::new_exact(value_class.clone()),
+                percentile_coercion.clone(),
+            ]));
+            variants.push(TypeSignature::Coercible(vec![
+                Coercion::new_exact(value_class),
+                percentile_coercion.clone(),
```

Comment:
I'm hesitant to accept a signature like this, since I don't like the look of this and it would need to be refactored anyway